driver-core/src/main/java/com/datastax/driver/core/Metadata.java

```diff
@@ -150,13 +170,73 @@ private void buildTableMetadata(KeyspaceMetadata ksm, List<Row> cfRows, Map<Stri
                         cols = Collections.<String, ColumnMetadata.Raw>emptyMap();
                     }
                 }
-                TableMetadata.build(ksm, cfRow, cols, cassandraVersion);
+
+                TableMetadata tm = TableMetadata.build(ksm, cfRow, cols, cassandraVersion);
+                TableMetadata tmPrevious = ksm.tables.get(tm.getName());
+                if (tmPrevious == null) {
+                    ksm.add(tm);
+                    triggerOnTableAdded(tm);
```

Comment:
:+1: .  In that case we should get rid of the code in Cluster#handle when handling a SCHEMA_CHANGE that runs it in cluster.manager.executor?